parlai/tasks/coco_caption/agents.py

```diff
@@ -79,6 +104,7 @@ def __init__(self, opt, version='2014'):
         self.num_epochs = self.opt.get('num_epochs', 0)
         self.image_loader = ImageLoader(opt)
         test_info_path, annotation_path, self.image_path = _path(opt, version)
+        self.candidates = load_candidates(opt['datapath'], version)
```

Comment:
it doesn't look like you use this anywhere, might need to add in the logic in `__getitem__`